core/src/main/java/io/methvin/watcher/DirectoryWatcher.java

```diff
@@ -275,14 +279,24 @@ public void watch() {
                 logger.debug(
                     "Failed to hash modified file [{}]. It may have been deleted.", childPath);
               }
-            } else {
-              onEvent(EventType.MODIFY, childPath, count, rootPath);
             }
           } else if (kind == ENTRY_DELETE) {
-            // we cannot tell if the deletion was on file or folder because path points nowhere
-            // (file/folder was deleted)
-            pathHashes.entrySet().removeIf(e -> e.getKey().startsWith(childPath));
-            onEvent(EventType.DELETE, childPath, count, rootPath);
+            // we cannot tell if this was a file or folder, but we can look at subpaths
+            if (fileHasher == null) {
+              // hashing is disabled, so just notify on the path we got the event for
+              onEvent(EventType.DELETE, childPath, count, rootPath);
+            } else {
+              // hashing is enabled, so we may know other paths that need to be deleted
+              Set<Path> subtreePaths =
```

Comment:
If hashing is enabled, there may be other paths in the map that have also been deleted as a consequence. Sometimes we'll get a delete for a directory but won't get a delete for files contained in the directory.